File: src/main/java/de/bst/five_cities/utils/DownloadUtils.java
package de.bst.five_cities.utils;

import static org.apache.commons.io.FilenameUtils.*;
import static org.apache.commons.io.IOUtils.*;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStream;
import java.net.URL;
import java.util.Date;

import org.apache.commons.io.FileUtils;

public class DownloadUtils {

	public static File getFile(URL source, File targetDir) throws IOException {
		File targetFile = new File(targetDir, getName(source.getPath()));
		if (targetFile.exists()) {
			notifySkippingOfDownload(targetFile);
			return targetFile;
		}

		return downloadFile(source, targetFile);
	}

	private static void notifySkippingOfDownload(File targetFile) {
		Date lastModified = new Date(targetFile.lastModified());
		Date now = new Date();
		double secondsOld = (now.getTime() - lastModified.getTime()) / 1000.0;
		System.out.println("Skipping download. File " + targetFile + " already exits and is " + secondsOld
				+ " seconds old (lastModified = " + lastModified + ")");
	}

	private static File downloadFile(URL source, File targetFile) throws IOException {
		FileUtils.forceMkdir(targetFile.getParentFile());
		InputStream input = source.openStream();
		FileWriter output = new FileWriter(targetFile);
		copy(input, output);
		input.close();
		output.close();

		return targetFile;
	}

}

File: src/main/scala/de/bst/five_cities/utils/DownloadUtils.scala
package de.bst.five_cities.utils

import java.io.{FileWriter, File}
import java.net.URL
import java.util.Date

import org.apache.commons.io.FileUtils
import org.apache.commons.io.FilenameUtils.getName
import org.apache.commons.io.IOUtils.copy

object DownloadUtils {
  def getFile(source: URL, targetDir: File): File = {
    val targetFile = new File(targetDir, getName(source.getPath))
    if (targetFile.exists) {
      notifySkippingOfDownload(targetFile)
      targetFile
    } else downloadFile(source, targetFile)
  }

  private def notifySkippingOfDownload(targetFile: File) {
    val lastModified = new Date(targetFile.lastModified)
    val now = new Date
    val secondsOld = (now.getTime - lastModified.getTime) / 1000.0
    println(s"Skipping download. File ${targetFile.getName} already exits and is $secondsOld seconds old $lastModified")
  }

  private def downloadFile(source: URL, targetFile: File): File = {
    FileUtils forceMkdir targetFile.getParentFile
    val input = source.openStream
    val output = new FileWriter(targetFile)
    copy(input, output)
    input.close
    output.close
    targetFile
  }
}

